### PR TITLE
Fix potential github action smells

### PR DESCRIPTION
**Pull request by ceddy4395**
Hey! 🙂
I want to contribute the following changes to your workflow:

- Avoid executing  scheduled workflows on forks
- Stop running workflows when there is a newer commit in branch

(These changes are part of a research Study at TU Delft looking at GitHub Action Smells. [Find out more](https://ceddy4395.github.io/research/gha-smells.html))

### CODE REVIEW

**Comment by ceddy4395** [comment]
@dotnet-policy-service agree

File: .github/workflows/whats-new.yml [hunk 1]
```diff
@@ -23,13 +23,14 @@ jobs:
     permissions:
       contents: write
       pull-requests: write
+    if: ${{ github.repository == 'dotnet/AspNetCore.Docs' }}
```
Comment (BillWagner) [inline .github/workflows/whats-new.yml:26]: I generally use the organization name only, so it's easier to reuse the code in multiple repos. (Yes, I know it's a copy / paste reuse, but still...)

```suggestion
    if: ${{ github.repository_owner == 'dotnet' }}
```

**Review by BillWagner — APPROVED** [review]
This looks good.  I had one suggestion, and then I'd say :shipit:

File: .github/workflows/whats-new.yml (hunk 1 above)
Comment (ceddy4395) [reply .github/workflows/whats-new.yml]: Ahh okee interesting! Thanks for the feedback :smile:

**Comment by Rick-Anderson** [comment]
@ceddy4395  thanks for the contribution.
@BillWagner  thanks for the quick review and suggestion.

**Pull request merged**